src/test/java/com/maxmind/db/DecoderTest.java

```diff
@@ -404,94 +398,77 @@ public void testArrays() throws IOException {
-                } else if (type.equals(Type.BOOLEAN)) {
-                    assertEquals(expect, decoder.decode(0, Boolean.class), desc);
-                } else {
-                    // We hit this for Type.MAP.
-
-                    Map got = decoder.decode(0, Map.class);
-                    Map expectMap = (Map) expect;
-
-                    assertEquals(expectMap.size(), got.size(), desc);
-
-                    for (Object keyObject : expectMap.keySet()) {
-                        String key = (String) keyObject;
-                        Object value = expectMap.get(key);
-
-                        if (value instanceof Object[]) {
-                            assertArrayEquals((Object[]) value, (Object[]) got.get(key), desc);
-                        } else {
-                            assertEquals(value, got.get(key), desc);
-                        }
+                assertEquals(expect, decoder.decode(0, BigInteger.class), desc);
+            } else if (type.equals(Type.DOUBLE)) {
+                assertEquals(expect, decoder.decode(0, Double.class), desc);
+            } else if (type.equals(Type.FLOAT)) {
+                assertEquals(expect, decoder.decode(0, Float.class), desc);
+            } else if (type.equals(Type.UTF8_STRING)) {
+                assertEquals(expect, decoder.decode(0, String.class), desc);
+            } else if (type.equals(Type.BOOLEAN)) {
+                assertEquals(expect, decoder.decode(0, Boolean.class), desc);
+            } else {
+                // We hit this for Type.MAP.
+
+                Map got = decoder.decode(0, Map.class);
+                Map expectMap = (Map) expect;
+
+                assertEquals(expectMap.size(), got.size(), desc);
+
+                for (Object keyObject : expectMap.keySet()) {
+                    String key = (String) keyObject;
+                    Object value = expectMap.get(key);
+
+                    if (value instanceof Object[]) {
+                        assertArrayEquals((Object[]) value, (Object[]) got.get(key), desc);
+                    } else {
+                        assertEquals(value, got.get(key), desc);
                     }
                 }
             }
         }
     }
 
-    /*
-     * I really didn't want to create temporary files for these tests, but it is
-     * pretty hard to abstract away from the file io system in a way that is
-     * Java 6 compatible
-     */
-    private static FileChannel getFileChannel(byte[] data) throws IOException {
-        File file = File.createTempFile(UUID.randomUUID().toString(), "tmp");
-        file.deleteOnExit();
-        RandomAccessFile raf = new RandomAccessFile(file, "rw");
-        FileChannel fc = raf.getChannel();
-        fc.write(ByteBuffer.wrap(data));
-        raf.close();
-        fc.close();
-        raf = new RandomAccessFile(file, "r");
-        return raf.getChannel();
+    private static ByteBuffer getByteBuffer(byte[] data) {
+        return ByteBuffer.wrap(data);
```

Comment:
Is this method really pulling its weight? It seems like it would be simpler to just replace the uses with `ByteBuffer.wrap`

Reply:
Fair point, fixed.